File: src/main/webapp/app/entities/invitations/invitation.ts
import {User} from "../../entities/user/user";
import {Organization} from "../../entities/organization/organization";

/**
 * This class represents an invitation for a user to join an organization.
 * */
export class Invitation {
    public invitedUser:User;
    public organization:Organization;
    public acceptId:string;
    public email:string;

    public accepted:boolean;
}

File: src/main/java/be/kdg/kandoe/backend/model/users/Invitation.java
package be.kdg.kandoe.backend.model.users;

import be.kdg.kandoe.backend.model.organizations.Organization;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.annotation.PostConstruct;
import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;
import java.util.Calendar;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
public class Invitation implements Serializable {
    @Id
    @GeneratedValue
    @Setter(value = AccessLevel.NONE)
    private Integer invitationId;

    @OneToOne(fetch = FetchType.EAGER)
    private User invitedUser;

    @OneToOne(fetch = FetchType.EAGER)
    private Organization organization;

    @Column(nullable = false, updatable = false, unique = true)
    private String acceptId;

    private boolean accepted;
}
